BraveRewardsUI/Extensions/BraveLedgerExtensions.swift

```diff
@@ -117,6 +120,90 @@ extension BraveLedger {
       })
     }
   }
+  
+  private var paymentId: String? {
+    var id: String?
+    rewardsInternalInfo { info in
+      id = info?.paymentId
+    }
+    return id
+  }
+  
+  public func setupDeviceCheckEnrollment(_ client: DeviceCheckClient, completion: @escaping () -> Void) {
+    // Enroll in DeviceCheck
+    client.generateToken { [weak self] (token, error) in
+      guard let self = self else { return }
+      if let error = error {
+        log.error("Failed to generate DeviceCheck token: \(error)")
+        completion()
+        return
+      }
+      let paymentId = self.paymentId ?? ""
+      client.generateEnrollment(paymentId: paymentId, token: token) { registration, error in
+        if let error = error {
+          log.error("Failed to enroll in DeviceCheck: \(error)")
+          completion()
+          return
+        }
+        guard let registration = registration else { return }
+        client.registerDevice(enrollment: registration) { error in
+          if let error = error {
+            log.error("Failed to register device with mobile attestation server: \(error)")
+            completion()
+            return
+          }
+        }
+      }
+    }
+  }
+  
+  public func claimPromotion(_ promotion: Promotion, completion: @escaping (_ success: Bool) -> Void) {
+    guard let paymentId = self.paymentId else { return }
+    let deviceCheck = DeviceCheckClient(environment: BraveLedger.environment)
+    let group = DispatchGroup()
+    if !DeviceCheckClient.isDeviceEnrolled() {
+      group.enter()
+      setupDeviceCheckEnrollment(deviceCheck) {
+        if !DeviceCheckClient.isDeviceEnrolled() {
```

Comment:
In the future since this is not a blocker, we can just check the callback of the enrollment for an error. If the callback returns nil, it’s successful. That way we don’t have to even check keychain. But for now it’s fine :)